File: Codechef/Codechef_JAN_LONG_2021/Watching_CPL.py
try:
    test = int(input())
    for _ in range(test):
        n, k = map(int, input().split())
        arr = list(map(int, input().split()))

        s = sum(arr) 
        if n == 1 or s < 2 * k:
            print(-1) 
            continue 
        else:
            min_sum = []  

            arr.sort(reverse = True) 

            flag = False 
            ans = 0

            min_sum.append(arr[0]) 
            temp_sum = arr[0] 

            for i in range(1, n):
                min_sum_temp = [] 
                temp_sum += arr[i] 

                for ele in min_sum:
                    t = [ele, arr[i], ele + arr[i]]
                    min_sum_temp += t 

                    if ele + arr[i] >= k:
                        if temp_sum - (ele + arr[i]) >= k:
                            
                            ans = i + 1
                            flag = True 
                            break 
                    if arr[i] >= k:
                        if temp_sum - arr[i] >= k:
                            ans = i + 1
                            flag = True 
                            break
                
                if(flag):
                    break
                
                min_sum = min_sum_temp.copy() 
            
            if(flag):
                print(ans) 
            else:
                print(-1)

except:
    pass

File: Codechef/2021/Codechef_JAN_LONG_2021/Watching_CPL.cpp
#include<bits/stdc++.h>
#define ll long long int
#define endl "\n"
using namespace std;

int main()
{
    ll test;
    cin >> test;
    while(test--)
    {
        ll n, i, j, k, ans = 0, sum = 0, count = 0;
        bool flag = false;
        cin >> n >> k;
        ll arr[n];
        vector<ll> min_sum;

        for(i = 0; i< n; i++)
        {
            cin >> arr[i];
            sum += arr[i];
        } 

        if(sum < 2 * k || n == 1)
        {
            cout << -1 << endl;
            continue;
        }

        sort(arr, arr + n, greater<ll>());

        min_sum.push_back(arr[0]);
        ans = arr[0];
        
        for(i = 1; i < n; i++)
        {
            vector<ll> min_sum_temp;
            ans += arr[i];

            for(j = 0; j < min_sum.size(); j++)
            {
                min_sum_temp.push_back(min_sum[j]);
                min_sum_temp.push_back(arr[i]);
                min_sum_temp.push_back(arr[i] + min_sum_temp[j]);

                if(min_sum[j] + arr[i] >= k)
                {
                    if(ans - (min_sum[j] + arr[i]) >= k)
                    {
                        count = i; 
                        flag = true;
                        break;
                    }
                }
                if(arr[i] >= k)
                {
                    if(ans - arr[i] >= k)
                    {
                        count = i;
                        flag = true;
                        break;
                    }
                }
            }
            if(flag)
            {
                break;
            }
            copy(min_sum_temp.begin(), min_sum_temp.end(), back_inserter(min_sum));
        }
        if(flag)
        {   
            if(count > n)
            {
                count = n;
            }
            cout << count + 1 << endl;
        }
        else  
        {
            cout << -1 << endl;
        }
    }
}